src/js/utils/rxjsUtils.js

```diff
@@ -2,18 +2,29 @@ import { timer } from "rxjs/observable/timer";
 
 import "rxjs/add/operator/delayWhen";
 import "rxjs/add/operator/scan";
+// import { Observable } from "rxjs/Observable";
+// import { Scheduler } from "rxjs/Scheduler";
 
 /* eslint-disable import/prefer-default-export */
+/**
+ * linearBackoff callback to use with e.g. delayWhen
+ *
+ * @param  {number} delay
+ * @param  {number} maxRetries
+ * @param  {number} [maxDelay=Number.MAX_SAFE_INTEGER]
+ * @param  {Scheduler} [scheduler=null]
+ * @return {Observable} notifier obervable
+ */
 export function linearBackoff(
-  maxRetries = 5,
-  delay = 1000,
+  delay,
+  maxRetries,
   maxDelay = Number.MAX_SAFE_INTEGER,
   scheduler = null
 ) {
   return errors =>
     errors
       .scan((count, error) => {
-        if (count >= maxRetries) {
+        if (maxRetries >= 0 && count >= maxRetries) {
```

Comment:
I would rather not increase count at all on this case by `return 0` everytime maxRetries < 0.

Reply:
but then there is no increase in backoff time? 🤔

Reply:
Yes, you are absolutely right. 

If you touch that again, can you change the name of `val`  to count on `delayWhen`. That would have helped me not make this mistake :D